module/canvas/hit-box.mjs

```diff
@@ -0,0 +1,76 @@
+/**
+ * A canvas layer which renders Crucible token hitboxes using the CrucibleHitBoxShader batch plugin.
+ * Displayed above the Grid layer and below the Tokens layer (zIndex 90).
+ */
+export default class CrucibleHitboxLayer extends foundry.canvas.layers.CanvasLayer {
+  constructor() {
+    super();
+    this.eventMode = "none";
+  }
+
+  /** @override */
+  static get layerOptions() {
+    return {name: "crucibleHitboxes", zIndex: 90};
+  }
+
+  /* ---------------------------------------- */
+
+  /**
+   * Collect the tokens whose hitboxes should be drawn.
+   * @returns {Token[]|null} Visible tokens if at least one is controlled, otherwise null
+   */
+  _collect() {
+    const anyControlled = canvas.tokens.controlled.length > 0;
+    if ( !anyControlled ) return null;
+    return canvas.tokens.placeables.filter(t => t?.visible);
```

Comment:
I don't really like that we loop twice here every frame, once to `filter` and again to iterate over the filtered list in `render`. 

If we want to design it this way it would be better for `_collect` to be a generator that yields each visible token one at a time or to just combine all of the logic together into a single loop in `render`.